src/Repository/ProductCommentRepository.php

```diff
@@ -166,6 +166,43 @@ public function getAverageGrade($productId, $validatedOnly)
         return (float) $qb->execute()->fetchColumn();
     }
 
+    /**
+     * @param array $productIds
+     * @param bool $validatedOnly
+     *
+     * @return array
+     */
+    public function getAverageGrades($productIds, $validatedOnly)
+    {
+        $sql = 'SELECT';
+
+        $count = count($productIds);
+
+        foreach ($productIds as $index => $id) {
+            $sql .= " SUM(IF(id_product = '" . $id . "' AND deleted = 0";
+            if ($validatedOnly) {
+                $sql .= " AND validate = '1'";
+            }
+            $sql .= " ,grade, 0))";
+            $sql .= " / SUM(IF(id_product = '" . $id . "' AND deleted = 0";
+            if ($validatedOnly) {
+                $sql .= " AND validate = '1'";
+            }
+            $sql .= " ,1, 0)) AS '" . $id . "'"; 
+
+            if ($count - 1 > $index) {
+                $sql .= ",";
+            }
+        }
+
+        $sql .= " FROM " . $this->databasePrefix . 'product_comment';
```

Comment:
```suggestion
        $sql .= ' FROM ' . $this->databasePrefix . 'product_comment';
```

Reply:
use quote instead of double quote everywhere. And use `pSQL` function to escape values

Reply:
I am not sure what you mean by escaping value with pSQL. Could you provide me some kind of reference to the topic.